src/mempool.rs

```diff
@@ -63,15 +62,28 @@ impl MempoolSyncUpdate {
         let to_add: Vec<Txid> = to_add.into_iter().collect();
         let mut new_entries = Vec::with_capacity(to_add.len());
 
-        for chunk in to_add.chunks(100) {
+        for txids_chunk in to_add.chunks(1000) {
             exit_flag.poll().context("mempool update interrupted")?;
-            let chunk_entries: Vec<Entry> = chunk
-                .par_iter()
-                .filter_map(|txid| -> Option<Entry> {
-                    // skip missing mempool entries
-                    let tx = daemon.get_transaction(txid, None).ok()?;
-                    let entry = daemon.get_mempool_entry(txid).ok()?;
-
+            let entries = daemon.get_mempool_entries(txids_chunk)?;
+            ensure!(
+                txids_chunk.len() == entries.len(),
+                "got {} mempools entries, expected {}",
+                entries.len(),
+                txids_chunk.len()
+            );
+            let txs = daemon.get_mempool_transactions(txids_chunk)?;
```

Comment:
It might be wise to chunk the raw transaction fetching depending on the size of each TX.

These days, with inscriptions running around, some TXs might be _much_ larger than others, so one batch could end up fetching several dozen megabytes worth of TX data, while another might only fetch a few hundred kilobytes. 

At this point in the code we can use the mempool entries to check how big each transaction is, which we can use to pace the TX fetching more evenly. This way we don't end up with some responses which are too large. 

The logic might go something like:

```python
all_raw_txs = []
while len(mempool_entries) > 0:
  fetch_bucket = []
  expected_response_size = 0
  for entry in mempool_entries:
    if entry.size + expected_response_size > MAX_RESP_SIZE:
      expected_response_size += entry.size
      mempool_entries.remove(entry)
      fetch_bucket.append(entry.txid)
    else:
      break
  raw_txs = daemon.fetch_txids_batch(fetch_bucket)
  all_raw_txs.extend(raw_txs)
```

@romanz I'd be happy to give this a go if you'd like. But this PR is already a huge improvement over the status quo, so I'd say it's safe to save this idea for a future PR.

Reply:
Thanks, good idea!
Let's implement it as a future PR.